cloud/provider.go

```diff
@@ -156,6 +165,7 @@ type Provider interface {
 	GetManagementPlatform() (string, error)
 	GetLocalStorageQuery(offset string) (string, error)
 	ExternalAllocations(string, string, string) ([]*OutOfClusterAllocation, error)
+	ApplyReservedInstancePricing(map[string]*Node)
```

Comment:
See notes in costmodel about this. It may be better to rename this to something more generic, like `ApplyNodePricing` where each provider can implement any custom node pricing. It also seems like moving the `getNodeCost` method to `provider.go` is justifiable.

Reply:
I like the `ApplyNodePricing` idea